test/workbox-background-sync/node/lib/test-Queue.mjs

```diff
@@ -0,0 +1,535 @@
+        `method that adds the failed request to the queue`, async function() {
+      sandbox.stub(Queue.prototype, 'addRequest');
+      const queue = new Queue('foo');
+      const plugin = queue.createPlugin();
+
+      plugin.fetchDidFail({request: new Request('/')});
+      expect(Queue.prototype.addRequest.calledOnce).to.be.true;
+      expect(Queue.prototype.addRequest.calledWith(
+          sinon.match.instanceOf(Request))).to.be.true;
+    });
+  });
+
+  describe(`addRequest`, function() {
+    it(`should serialize the request and store it in IndexedDB`,
+        async function() {
+      sandbox.spy(DBWrapper.prototype, 'add');
+      sandbox.stub(self.registration, 'active').value({});
+
+      const queue = new Queue('foo');
+      const requestUrl = 'https://example.com';
+      const requestInit = {
+        method: 'POST',
+        body: 'testing...',
+        headers: {'x-foo': 'bar'},
+        mode: 'cors',
+      };
+      const request = new Request(requestUrl, requestInit);
+
+      await queue.addRequest(request);
+
+      expect(DBWrapper.prototype.add.calledOnce).to.be.true;
+      expect(DBWrapper.prototype.add.calledWith({
+        queueName: 'foo',
+        storableRequest: sinon.match({
+          url: requestUrl,
+          timestamp: sinon.match.number,
+          requestInit: sinon.match(requestInit),
+        }),
+      })).to.be.true;
+    });
+
+    it(`should register to receive sync events for a unique tag`,
+        async function() {
+      sandbox.stub(self.registration, 'active').value({});
+      sandbox.stub(self.registration, 'sync').value({
+        register: sinon.stub().resolves(),
+      });
+
+      const queue = new Queue('foo');
+      const requestUrl = 'https://example.com';
+      const requestInit = {
+        method: 'POST',
+        body: 'testing...',
+        headers: {'x-foo': 'bar'},
+        mode: 'cors',
+      };
+      const request = new Request(requestUrl, requestInit);
+
+      await queue.addRequest(request);
+
+      expect(self.registration.sync.register.calledOnce).to.be.true;
+      expect(self.registration.sync.register.calledWith(
+          'workbox-background-sync:foo')).to.be.true;
+    });
+
+    it(`should delay registration until the SW is active`, async function() {
+      sandbox.stub(self.registration, 'active').value(undefined);
+      sandbox.stub(self.registration, 'sync').value({
+        register: sinon.spy(),
+      });
+
+      const queue = new Queue('foo');
+      const requestUrl = 'https://example.com';
+      const requestInit = {
+        method: 'POST',
+        body: 'testing...',
+        headers: {'x-foo': 'bar'},
+        mode: 'cors',
+      };
+      const request = new Request(requestUrl, requestInit);
+
+      // This shouldn't error, even though activation hasn't happened yet.
+      await queue.addRequest(request);
+
+      // Allow time to ensure async registration didn't happen.
+      await sleep(100);
```

Comment:
This feels flakey and unnecessary. This doesn't seem to affect the flow at all of this test. If you do need timeouts please use sinon fake timers.